extensions/narayana-jta/runtime/src/main/java/io/quarkus/narayana/jta/runtime/NotifyingUserTransaction.java

```diff
@@ -0,0 +1,72 @@
+package io.quarkus.narayana.jta.runtime;
+
+import javax.transaction.HeuristicMixedException;
+import javax.transaction.HeuristicRollbackException;
+import javax.transaction.NotSupportedException;
+import javax.transaction.RollbackException;
+import javax.transaction.SystemException;
+import javax.transaction.UserTransaction;
+
+import org.jboss.logging.Logger;
+
+import com.arjuna.ats.internal.jta.transaction.arjunacore.TransactionImple;
+
+public class NotifyingUserTransaction extends TransactionScopedNotifier implements UserTransaction {
+
+    private static final Logger LOG = Logger.getLogger(NotifyingUserTransaction.class);
+
+    private final UserTransaction delegate;
+
+    public NotifyingUserTransaction(UserTransaction delegate) {
+        this.delegate = delegate;
+    }
+
+    @Override
+    public void begin() throws NotSupportedException, SystemException {
+        delegate.begin();
+        initialized(TransactionImple.getTransaction().toString());
+    }
+
+    @Override
+    public void commit() throws RollbackException, HeuristicMixedException, HeuristicRollbackException, SecurityException,
+            IllegalStateException, SystemException {
+        String id = TransactionImple.getTransaction().toString();
+        beforeDestroyed(id);
+        try {
+            delegate.commit();
+        } finally {
+            destroyed(id);
+        }
+    }
+
+    @Override
+    public void rollback() throws IllegalStateException, SecurityException, SystemException {
+        String id = TransactionImple.getTransaction().toString();
```

Comment:
@mmusgrov I hope that this the correct way to obtain the transaction id ;-)

Reply:
Yes, @mkouba , that get's the currently bound transaction context.

BTW if there is no active transaction then `TransactionImple.getTransaction().toString()` will fail with a NPE. So `throw new SystemException("The transaction is not active!");` would be preferable.

The same comment applies to the other usages (the commit method and the commit/rollback methods in `NotifyingTransactionManager.java`).

Reply:
Done. @mmusgrov pls review.